### PR TITLE
add rsa patch (compatibility for building with newer versions of openssl on host)

### PR DESCRIPTION
**Pull request by blondejamtart**

### CODE REVIEW

**Comment by thomascobb** [comment]
Where did you get the patch from, or did you write it yourself?

**Comment by blondejamtart** [comment]
Patch was copied from a similar commit in openwrt

File: Makefile [hunk 1]
```diff
@@ -7,8 +7,10 @@
 # Note: if these files have been downloaded through the releases directory then
 # they need to be renamed with the appropriate {u-boot,linux}-xlnx- prefix so
 # that the file name and contents match.
-MD5_SUM_u-boot-xlnx-xilinx-v2015.1 = b6d212208b7694f748727883eebaa74e
-MD5_SUM_linux-xlnx-xilinx-v2015.1  = 930d126df2113221e63c4ec4ce356f2c
+MD5_SUM_u-boot-xlnx-xilinx-v2015.1 	= b6d212208b7694f748727883eebaa74e
```
Comment (Araneidae) [inline Makefile:10]: Don't change the whitespace on the existing MD5_SUM lines, so they don't appear in the diff

**Comment by Araneidae** [comment]
Can you reference the source of the patch please?

**Comment by blondejamtart** [comment]
Patch copied from here:
https://git.openwrt.org/?p=openwrt/openwrt.git;a=blob;f=tools/mkimage/patches/210-openssl-1.1.x-compat.patch;h=fa7c99f39b0a65f0d784473ca9b8fde836e4fa6e;hb=70b104f98c0657323b28fce140b73a94bf3eb756

**Comment by Araneidae** [comment]
Ta.  What you can do is drop the commit text for that patch:
```
tools: mkimage: fix build with OpenSSL 1.1.x (FS#182)

The OpenSSL 1.1.x version series undergone some major API changes which made
the RSA structure opaque and deprecated a number of methods, so add some
conditional compat code to make the u-boot source build again.

Signed-off-by: Jo-Philipp Wich <jo@mein.io>
```
directly into the top of the patch (`patch` helpfully ignores anything not prefixed with the appropriate preamble) and possibly have a concise version of the link in your commit message.

**Comment by Araneidae** [comment]
Splendid!

Do you want to rebase all your commits into a single commit?

**Pull request merged**